crates/turborepo-scm/src/package_deps.rs

```diff
@@ -2,13 +2,28 @@ use std::collections::HashMap;
 
 use itertools::{Either, Itertools};
 use tracing::debug;
-use turbopath::{AbsoluteSystemPath, AnchoredSystemPathBuf, PathError, RelativeUnixPathBuf};
+use turbopath::{
+    AbsoluteSystemPath, AnchoredSystemPath, AnchoredSystemPathBuf, PathError, RelativeUnixPathBuf,
+};
 
 use crate::{hash_object::hash_objects, Error, Git, SCM};
 
 pub type GitHashes = HashMap<RelativeUnixPathBuf, String>;
 
 impl SCM {
+    pub fn get_hashes_for_files(
+        &self,
+        turbo_root: &AbsoluteSystemPath,
+        files: &[impl AsRef<AnchoredSystemPath>],
```

Comment:
Made this AsRef because we have a slice of `AnchoredSystemPathBuf`'s in global hash